### PR TITLE
fix(compiler): Respect stack type when dropping statements

### PR DESCRIPTION
**Pull request by ospencer**

### CODE REVIEW

**Review by marcusroberts — APPROVED** [review]
I can't say I can read the snapshots very well but the code change looks good.

**Review by peblair — APPROVED** [review]

File: compiler/src/codegen/transl_anf.re [hunk 1]
```diff
@@ -260,8 +260,8 @@ module RegisterAllocation = {
       | MSet(b, i) =>
         MSet(apply_allocation_to_bind(b), apply_allocations(ty, allocs, i))
       | MAllocate(x) => MAllocate(x)
-      | MDrop(i) => MDrop(apply_allocations(ty, allocs, i))
-      | MIncRef(i) => MDrop(apply_allocations(ty, allocs, i))
+      | MDrop(i, a) => MDrop(apply_allocations(ty, allocs, i), a)
+      | MIncRef(i) => MIncRef(apply_allocations(ty, allocs, i))
```
Comment (phated) [inline compiler/src/codegen/transl_anf.re:264]: Was this a typo bug previously?

**Review by phated — APPROVED** [review]
Nice! Skimmed the snapshots too

File: compiler/src/codegen/transl_anf.re (hunk 1 above)
Comment (ospencer) [reply compiler/src/codegen/transl_anf.re:264]: Yes 🤦 

But all of this code is completely unused anyway.

**Pull request merged**